parsers/IS.py

```diff
@@ -0,0 +1,105 @@
+import requests
+
+# please try to write PEP8 compliant code (use a linter). One of PEP8's
+# requirement is to limit your line length to 79 characters.
+
+
+def fetch_production(
+    zone_key="IS",
+    session=None,
+    target_datetime: datetime.datetime = None,
+    logger: logging.Logger = logging.getLogger(__name__),
+):
+    """Requests the last known production mix (in MW) of a given country
+
+    Arguments:
+    ----------
+    zone_key: used in case a parser is able to fetch multiple countries
+    session: request session passed in order to re-use an existing session
+    target_datetime: the datetime for which we want production data. If not
+      provided, we should default it to now. If past data is not available,
+      raise a NotImplementedError. Beware that the provided target_datetime is
+      UTC. To convert to local timezone, you can use
+      `target_datetime = arrow.get(target_datetime).to('America/New_York')`.
+      Note that `arrow.get(None)` returns UTC now.
+    logger: an instance of a `logging.Logger` that will be passed by the
+      backend. Information logged will be publicly available so that correct
+      execution of the logger can be checked. All Exceptions will automatically
+      be logged, so when something's wrong, simply raise an Exception (with an
+      explicit text). Use `logger.warning` or `logger.info` for information
+      that can useful to check if the parser is working correctly. A default
+      logger is used so that logger output can be seen when coding / debugging.
+
+    Returns:
+    --------
+    If no data can be fetched, any falsy value (None, [], False) will be
+      ignored by the backend. If there is no data because the source may have
+      changed or is not available, raise an Exception.
+
+    A dictionary in the form:
+    {
+      'zoneKey': 'IS',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
+      },
+      'storage': {
+          'hydro': -10.0,
+      },
+      'source': 'mysource.com'
+    }
+    """
+    r = session or requests.session()
+    if target_datetime is None:
+        url = "https://amper.landsnet.is/generation/api/Values"
+    else:
+        # WHEN HISTORICAL DATA IS NOT AVAILABLE
+        raise NotImplementedError("This parser is not yet able to parse past dates")
+
+    res = r.get(url)
+    assert res.status_code == 200, (
+        "Exception when fetching production for "
+        "{}: error when calling url={}".format(zone_key, url)
+    )
+
+    obj = res.json()
+    data = {
+        "zoneKey": zone_key,
+        "production": {},
+        "storage": {},
+        "source": "https://amper.landsnet.is/generation/api/Values",
```

Comment:
We typically only put the domain here:
```suggestion
        "source": "amper.landsnet.is",
```

or 
```suggestion
        "source": "landsnet.is",
```